src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+static uint8_t *jetiExBusFrame;
+
+
+static uint16_t jetiExBusChannelData[JETIEXBUS_CHANNEL_COUNT];
+static void jetiExBusDataReceive(uint16_t c);
+static uint16_t jetiExBusReadRawRC(rxRuntimeConfig_t *rxRuntimeConfig, uint8_t chan);
+
+
+#ifdef TELEMETRY
+
+static uint8_t jetiExBusTelemetryFrame[JETIEXBUS_MAX_FRAME_SIZE];
+static uint8_t jetiExBusTransceiveState = EXBUS_TRANS_RX;
+static uint8_t *jetiExTelemetry=&jetiExBusTelemetryFrame[EXBUS_HEADER_DATA];
+static void sendJetiExBusTelemetry(uint8_t packetID);
+
+uint8_t updateCRC8( uint8_t crc, uint8_t data );
+uint8_t calcCRC8(uint8_t *pt, uint8_t msgLen);
+
+#endif //TELEMETRY
+
+uint16_t updateCRC16( uint16_t crc, uint8_t data );
+uint16_t calcCRC16(uint8_t *pt, uint8_t msgLen);
+
+
+bool jetiExBusInit(rxConfig_t *rxConfig, rxRuntimeConfig_t *rxRuntimeConfig, rcReadRawDataPtr *callback)
+{
+    UNUSED(rxConfig);
+    
+    if (callback) {
+        *callback = jetiExBusReadRawRC;
+    }
+
+    rxRuntimeConfig->channelCount = JETIEXBUS_CHANNEL_COUNT;
+    jetiExBusFramePosition = 0;
+    jetiExBusFrameLength = JETIEXBUS_MAX_FRAME_SIZE;
+
+    portConfig = findSerialPortConfig(FUNCTION_RX_SERIAL);
+    
+    if (!portConfig) {
+        return false;
+    }
+    
+    jetiExBusPort = openSerialPort(portConfig->identifier, FUNCTION_RX_SERIAL, jetiExBusDataReceive, JETIEXBUS_BAUDRATE, MODE_RXTX, JETIEXBUS_OPTIONS );
+    serialSetMode(jetiExBusPort, MODE_RX);
+    return jetiExBusPort != NULL;
+}
+
+
+// Jeti Ex Bus CRC calculations for single byte
+uint16_t updateCRC16( uint16_t crc, uint8_t data )
+{
+    uint16_t ret_val;
+    data ^= (uint8_t)(crc) & (uint8_t)(0xFF);
+    data ^= data << 4;
+    ret_val = ((((uint16_t)data << 8) | ((crc & 0xFF00) >> 8))
+               ^ (uint8_t)(data >> 4)
+               ^ ((uint16_t)data << 3));
+    return ret_val;
+} 
+
+
+// Jeti Ex Bus CRC calculations for a frame
+uint16_t calcCRC16(uint8_t *pt, uint8_t msgLen)
+{
+    uint16_t crc16_data = 0;
+    uint8_t mlen = 0;
+    while(mlen < msgLen) {
+        crc16_data = updateCRC16(crc16_data, pt[mlen]);
+        mlen++;
+    }
+    return(crc16_data);
+}
+
+#ifdef TELEMETRY
+
+// Jeti Ex Telemetry CRC calculations for single byte
+uint8_t updateCRC8( uint8_t crc, uint8_t data )
+{
+    crc ^= data;
+    return (crc ^ (crc << 1) ^ (crc << 2) ^ (0x0e090700 >> ((crc >> 3) & 0x18)));
+} 
+
+// Jeti Ex Telemetry CRC calculations for a frame
+uint8_t calcCRC8(uint8_t *pt, uint8_t msgLen)
+{
+    uint8_t crc8_data = 0;
+    uint8_t mlen = 0;
+    for (mlen = 0; mlen < msgLen; mlen++) {
+        crc8_data = updateCRC8(crc8_data, pt[mlen]);
+    }
+    return(crc8_data);
+}
+
+#endif //TELEMETRY
+
+
+void jetiExBusDecodeFrame(uint8_t *pFrame)
+{
+    uint16_t value;
+    uint8_t frameAddr;
+
+    // Decode header
+    switch (((uint16_t)jetiExBusFrame[EXBUS_HEADER_SYNC] << 8) + ((uint16_t)jetiExBusFrame[EXBUS_HEADER_REQ])){
```

Comment:
it is a bit better to use bit-or `|` instead of + .. the result is identical, but `|` is easier to read